src/service/exception_handlers.py

```diff
@@ -0,0 +1,82 @@
+"""
+Exception handlers for the FastAPI application.
+"""
+
+import logging
+
+from fastapi import HTTPException, Request, status
+from fastapi.exceptions import RequestValidationError
+from fastapi.responses import JSONResponse
+
+from src.service import errors
+from src.service.error_mapping import map_error
+from src.service.exceptions import SparkManagerError
+
+logger = logging.getLogger(__name__)
+
+
+def _format_error(
+    status_code: int,
+    error_code: int | None,
+    error_type_str: str | None,
+    message: str | None,
+):
+    """Format error response with consistent structure."""
+    return JSONResponse(
+        status_code=status_code,
+        content={
+            "error": error_code,
+            "error_type": error_type_str,
+            "message": message or error_type_str or "Unknown error",
+        },
+    )
+
+
+async def universal_error_handler(request: Request, exc: Exception):
+    """
+    Universal handler for all types of exceptions.
+
+    Handles:
+    - SparkManagerError and its subclasses (including all specific error types)
+    - HTTPException from FastAPI
+    - RequestValidationError for request validation
+    - Generic exceptions
+    """
+    # Default values
+    error_code = None
+    error_type_str = None
+    status_code = status.HTTP_500_INTERNAL_SERVER_ERROR
+    message = None
+
+    if isinstance(exc, SparkManagerError):
+        # handle SparkManagerError and subclasses
+        error_type, status_code = map_error(exc)
+
+        # Extract values from error_type if available
+        if error_type:
+            error_code = error_type.error_code
+            error_type_str = error_type.error_type
+        else:
+            raise ValueError("this should be impossible")
```

Comment:
```suggestion
```
I just realized this wrong because SparkManagerException can return no error type.

Reply:
Yeah, the error type string should be for the specific codes from the error mapping, so a user can respond to the error based on either that enumeration or None

Reply:
The idea is that if they get one of those error codes in the enum there's something they could potentially do to fix thing. If there's no code there's nothing they can do

Reply:
👍